terraform/main.tf

```diff
@@ -396,3 +396,47 @@ resource "google_service_account_iam_member" "gce-default-account-iam" {
   role               = "roles/iam.serviceAccountUser"
   member             = "serviceAccount:345950277072@cloudbuild.gserviceaccount.com"
 }
+
+resource "google_secret_manager_secret" "pudl_usage_metrics_db_connection_string" {
+  secret_id = "pudl-usage-metrics-db-connection-string"
+  replication {
+    auto {}
+  }
+}
+
+resource "google_storage_bucket" "pudl_usage_metrics_archive_bucket" {
+  name          = "pudl-usage-metrics-archives.catalyst.coop"
+  location      = "US"
+  storage_class = "STANDARD"
+
+  uniform_bucket_level_access = true
+}
+
+resource "google_service_account" "usage_metrics_archiver" {
+  account_id   = "usage-metrics-archiver"
+  display_name = "PUDL usage metrics archiver github action service account"
+}
+
+resource "google_storage_bucket_iam_member" "usage_metrics_archiver_gcs_iam" {
+  for_each = toset(["roles/storage.objectCreator", "roles/storage.objectViewer"])
+
+  bucket = google_storage_bucket.pudl_usage_metrics_archive_bucket.name
+  role = each.key
+  member = "serviceAccount:${google_service_account.usage_metrics_archiver.email}"
+}
+
+resource "google_storage_bucket_iam_member" "usage_metrics_etl_gcs_iam" {
+  for_each = toset(["roles/storage.legacyBucketReader", "roles/storage.objectViewer"])
+
+  bucket = google_storage_bucket.pudl_usage_metrics_archive_bucket.name
+  role = each.key
+  member = "serviceAccount:pudl-usage-metrics-etl@catalyst-cooperative-pudl.iam.gserviceaccount.com"
+}
+
+resource "google_storage_bucket_iam_member" "usage_metrics_etl_s3_logs_gcs_iam" {
+  for_each = toset(["roles/storage.legacyBucketReader", "roles/storage.objectViewer"])
+
+  bucket = "pudl-s3-logs.catalyst.coop"
```

Comment:
Should we manage this bucket via TF too?

Reply:
Probably! How can we manage a resource in terraform that has already been creating in the UI? Also, we should probably move the contents of `pudl-s3-logs.catalyst.coop` to `pudl-usage-metrics-archives.catalyst.coop/s3/` for consistency.

Reply:
Here's the [documentation](https://developer.hashicorp.com/terraform/tutorials/state/resource-drift#introduce-drift) about handling "resource drift"!

Reply:
I think we probably don't need the `pudl` part because the bucket lives in the `pudl-catalyst-cooperative` GCP project. It would be nice to keep `archive` because I could imagine having a bucket to store the outputs of our usage metric ETL to parquet files if we ever move to Big Query. How about `usage-metrics-archive.catalyst.coop`?

Reply:
@bendnorman Would love to coordinate a name change with the flight of PRs in `pudl-usage-metrics` as much as possible, so just let me know what you're thinking.

Reply:
I think we'll keep it as is for now but I created an issue to rename [it down the line](https://github.com/catalyst-cooperative/pudl-usage-metrics/issues/170).